src/alpaka/AlpakaCore/alpakaWorkDivHelper.h

```diff
@@ -320,6 +320,495 @@ namespace cms {
       return isNextStrideElementValid;
     }
 
+    /*
+     * Class which simplifies "for" loops over elements index 
+     */
+    template <typename T, typename T_Acc>
+    class elements_with_stride {
+    public:
+
+    ALPAKA_FN_ACC elements_with_stride(const T_Acc& acc) {
+            const unsigned int dimIndex = 0;
+            const Idx threadIdxLocal(alpaka::getIdx<alpaka::Block, alpaka::Threads>(acc)[dimIndex]);
+            const Idx blockIdxInGrid(alpaka::getIdx<alpaka::Grid, alpaka::Blocks>(acc)[dimIndex]);
+
+            const Idx blockDimension(alpaka::getWorkDiv<alpaka::Block, alpaka::Elems>(acc)[dimIndex]);
+            const Idx gridDimension(alpaka::getWorkDiv<alpaka::Grid, alpaka::Elems>(acc)[dimIndex]);
+
+            thread_ = blockDimension * blockIdxInGrid + threadIdxLocal;
+            stride_ = blockDimension * gridDimension;
+            extent_ = stride_;
+            blockDim = blockDimension;
```

Comment:
Are all these missing the loop over elements of a thread? (or am I missing something?)

Reply:
This was understood in the last comment, right?